scripts/signup.js

```diff
@@ -1,125 +1,218 @@
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ email }),
+    });
 
-    // 비밀번호 입력부분이 비어 있는 경우
-    if (!passwordValue) {
-      passwordError.textContent = "비밀번호를 입력해 주세요.";
-      passwordInput.classList.add("invalid");
-      // 비밀번호 입력부분이 비어있지 않는 경우 + 비밀번호 형식이 유효하지 않은 경우
-    } else if (!isValidPassword(passwordValue)) {
-      passwordError.textContent =
-        "비밀번호는 영문, 숫자 조합 8자 이상 입력해 주세요.";
-      passwordInput.classList.add("invalid");
-      // 비밀번호 입력부분이 비어있지 않은 경우 + 비밀번호 형식이 유효한 형식인 경우
-    } else {
-      passwordError.textContent = "";
-      passwordInput.classList.remove("invalid");
+    // 응답코드 200 : 중복되지 않은 이메일 확인
+    if (response.status === HTTP_STATUS_OK) {
+      const data = await response.json();
+      return data.isDuplicated;
     }
-
-    // 비밀번호 확인 입력부분이 비어 있는 경우
-    if (!confirmPasswordValue) {
-      confirmPasswordError.textContent = "비밀번호를 입력해 주세요.";
-      confirmPasswordInput.classList.add("invalid");
-      // 비밀번호와 비밀번호 확인이 일치하지 않는 경우
-    } else if (passwordValue !== confirmPasswordValue) {
-      confirmPasswordError.textContent = "비밀번호가 일치하지 않아요.";
-      confirmPasswordInput.classList.add("invalid");
-      // 비밀번호 확인 입력부분이 비어있지 않은 경우 + 비밀번호 형식이 유효한 형식인 경우
-    } else {
-      confirmPasswordError.textContent = "";
-      confirmPasswordInput.classList.remove("invalid");
+    // 응답코드 409 : 사용할 수 없는 중복된 이메일 오류
+    else if (response.status === HTTP_STATUS_CONFLICT) {
+      return true;
+    }
+    // 그외 응답코드
+    else {
+      throw new Error("서버 오류");
     }
+  } catch (error) {
+    console.error("이메일 중복 확인 오류: ", error.message);
+    return false;
   }
+}
 
-  emailInput.addEventListener("blur", handleEmailInput);
-  emailInput.addEventListener("input", handleEmailInput);
-  passwordInput.addEventListener("blur", handlePasswordInput);
-  passwordInput.addEventListener("input", handlePasswordInput);
-  confirmPasswordInput.addEventListener("blur", handlePasswordInput);
-  confirmPasswordInput.addEventListener("input", handlePasswordInput);
-
-  function validateSignUp() {
-    const emailError = document.querySelector("#emailError").textContent;
-    const passwordError = document.querySelector("#passwordError").textContent;
-    const confirmPasswordError = document.querySelector(
-      "#confirmPasswordError"
-    ).textContent;
-
-    return !emailError && !passwordError && !confirmPasswordError;
+// 비밀번호 입력 처리 함수 : 유효성 검사 및 에러 메시지 관리
+function handlePasswordInput(event) {
+  const passwordValue = passwordInput.value;
+  const confirmPasswordValue = confirmPasswordInput.value;
+
+  // 현재 입력 필드에 따라 에러 메시지 지우기
+  if (event.target === passwordInput) {
+    passwordError.textContent = "";
+    passwordInput.classList.remove(INVALID_CLASS);
+  } else if (event.target === confirmPasswordInput) {
+    confirmPasswordError.textContent = "";
+    confirmPasswordInput.classList.remove(INVALID_CLASS);
```

Comment:
event.target이 passwordInput이나 confirmPasswordInput이 아닌 경우가 없다면 아래처럼 작성해 중복을 줄일 수 있을 것 같아요

```suggestion
    event.target.textContent = "";
    event.target.classList.remove(INVALID_CLASS);
```